lib/wanda/execution/server.ex

```diff
@@ -80,9 +80,24 @@ defmodule Wanda.Execution.Server do
   @impl true
   def handle_info(
         :timeout,
-        %State{} = state
+        %State{
+          execution_id: execution_id,
+          group_id: group_id,
+          gathered_facts: gathered_facts,
+          targets: targets,
+          checks: checks,
+          agents_gathered: agents_gathered
+        } = state
       ) do
-    :ok = Messaging.publish("results", :timeout)
+    targets =
+      Enum.filter(targets, fn %Target{agent_id: agent_id} ->
+        agent_id not in agents_gathered
+      end)
+
+    gathered_facts = Gathering.put_gathering_timeouts(gathered_facts, targets)
+    result = Evaluation.execute(execution_id, group_id, checks, gathered_facts)
```

Comment:
This will most probably fail. We fixed this using:
`execution_completed = Messaging.Mapper.to_execution_completed(result)` and publishing that variable

Reply:
Ah yeah you're right, my bad